src/wallet/wallet.cpp

```diff
@@ -4333,21 +4333,18 @@ bool CWallet::TopUpKeyPool(unsigned int kpSize)
             if (i < missingInternal) {
                 fInternal = true;
             }
-            if (!setInternalKeyPool.empty()) {
-                nEnd = *(--setInternalKeyPool.end()) + 1;
-            }
-            if (!setExternalKeyPool.empty()) {
-                nEnd = std::max(nEnd, *(--setExternalKeyPool.end()) + 1);
-            }
-            // TODO: implement keypools for all accounts?
-            if (!walletdb.WritePool(nEnd, CKeyPool(GenerateNewKey(walletdb, 0, fInternal), fInternal))) {
```

Comment:
I see you fixed the argument discrepancy in a later fix commit, but you omitted the TODO comment that you also removed by accident. We should either re-add that comment or figure out how to deal with multi-account derivation. My current assumption is that we can actually ignore multi-accounts and always use 0 (and thus change the GenerateNewKey method).